presto-oracle/src/test/java/io/prestosql/plugin/oracle/TestOracleDistributedQueries.java

```diff
@@ -371,6 +357,7 @@ public void testInsertWithCoercion()
         assertUpdate("INSERT INTO test_insert_with_coercion (char_column, bounded_varchar_column, unbounded_varchar_column) VALUES (CAST('aa     ' AS varchar), CAST('aa     ' AS varchar), CAST('aa     ' AS varchar))", 1);
         assertUpdate("INSERT INTO test_insert_with_coercion (char_column, bounded_varchar_column, unbounded_varchar_column) VALUES (NULL, NULL, NULL)", 1);
         assertUpdate("INSERT INTO test_insert_with_coercion (char_column, bounded_varchar_column, unbounded_varchar_column) VALUES (CAST(NULL AS varchar), CAST(NULL AS varchar), CAST(NULL AS varchar))", 1);
+        // TODO Reimplement because this assumes DATE and TIMESTAMP are different types, but in Oracle connector DATE is mapped to TIMESTAMP
```

Comment:
Oracle's DATE and Oracle's  TIMESTAMP are different types

are you referring to Presto's DATE and Presto's TIMESTAMP?

Reply:
We have mapped Oracle's DATE to Presto's TIMESTAMP. So coercion of TIMESTAMP into DATE during insertion won't actually lead to a coercion being performed I think. See https://github.com/prestosql/presto/blob/master/presto-oracle/src/test/java/io/prestosql/plugin/oracle/OracleDataTypes.java#L223-L228.

Reply:
Makes sense. I also don't see any way to re-implement without changing the type mapping (for no gains apart from this test).